File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_ai_hub/lib/src/state/ai_hub_state.dart
import '../models/ai_hub_models.dart';

enum AiHubStatus { idle, loading, ready, error }

class AiHubState {
  const AiHubState({
    this.status = AiHubStatus.idle,
    this.lastResult,
    this.error,
  });

  final AiHubStatus status;
  final AiHubPageResult<Object?>? lastResult;
  final String? error;

  AiHubState loading() => const AiHubState(status: AiHubStatus.loading);

  AiHubState loaded(AiHubPageResult<Object?> result) =>
      AiHubState(status: AiHubStatus.ready, lastResult: result);

  AiHubState failed(String message) =>
      AiHubState(status: AiHubStatus.error, error: message);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-ai-hub/src/state/aiHubState.ts
import type { AiHubPageResult } from "../types/aiHubModels.js";

export interface AiHubState {
  readonly status: "idle" | "loading" | "ready" | "error";
  readonly lastResult?: AiHubPageResult<unknown>;
  readonly error?: string;
}

export function initialAiHubState(): AiHubState {
  return { status: "idle" };
}

export function reduceAiHubState(
  state: AiHubState,
  event:
    | { type: "load" }
    | { type: "loaded"; result: AiHubPageResult<unknown> }
    | { type: "failed"; error: string },
): AiHubState {
  switch (event.type) {
    case "load":
      return { ...state, status: "loading", error: undefined };
    case "loaded":
      return { status: "ready", lastResult: event.result };
    case "failed":
      return { ...state, status: "error", error: event.error };
  }
}
